### ISSUE TITLE
mpileup -C requires at least 3 readings.

### ISSUE THREAD

**Issue by jkbonfield** [samtools/samtools#125]
The mpileup -C parameter only takes effect on the 3rd reading onwards in a sam or bam file.

**Comment by jkbonfield** [comment]
Having reviewed this again, I'm amazed it's not also taking into account the date of Easter and the phase of the moon.  Never before has there been such a huge disparity behind what an argument actually does and what the man page says!

**Comment by jkbonfield** [comment]
Here is an example of the -C error:

```
jkb@seq3b[samtools.../samtools] cat x.ref
>x
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
jkb@seq3b[samtools.../samtools] cat x.sam
@HD     VN:1.0  SO:coordinate
@SQ     SN:x    LN:100
s1      67      x       1       40      50M     *       0       0       TCTCTGGGGTCTCACCCACGACCAACTCCCTGGGCCTGGCACCAGGGAGC       HIKLLKJKHDELMJHFKIKDKDKIHHILFKLHBMKGLJ7KGDJHBBFFGD      BQ:Z:IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII MD:Z:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
s2      67      x       2       40      50M     *       0       0       TCTCTGGGGTCTCACCCACGACCAACTCCCTGGGCCTGGCACCAGGGAGC       HIKLLKJKHDELMJHFKIKDKDKIHHILFKLHBMKGLJ7KGDJHBBFFGD      BQ:Z:IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII MD:Z:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
s2      67      x       3       40      50M     *       0       0       TCTCTGGGGTCTCACCCACGACCAACTCCCTGGGCCTGGCACCAGGGAGC       HIKLLKJKHDELMJHFKIKDKDKIHHILFKLHBMKGLJ7KGDJHBBFFGD      BQ:Z:IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII MD:Z:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
s4      67      x       4       40      50M     *       0       0       TCTCTGGGGTCTCACCCACGACCAACTCCCTGGGCCTGGCACCAGGGAGC       HIKLLKJKHDELMJHFKIKDKDKIHHILFKLHBMKGLJ7KGDJHBBFFGD      BQ:Z:IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII MD:Z:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
s5      67      x       5       40      50M     *       0       0       TCTCTGGGGTCTCACCCACGACCAACTCCCTGGGCCTGGCACCAGGGAGC       HIKLLKJKHDELMJHFKIKDKDKIHHILFKLHBMKGLJ7KGDJHBBFFGD      BQ:Z:IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII MD:Z:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
jkb@seq3b[samtools.../samtools] ./samtools mpileup -B -q0 -Q0 -A -f x.ref x.sam 2>/dev/null | egrep '\^'
x       1       A       1       ^IT     H
x       2       A       2       C^IT    IH
x       3       A       3       TC^IT   KA!
x       4       A       4       CTC^IT  LB!H
x       5       A       5       TCTC^IT LC!IH
jkb@seq3b[samtools.../samtools] ./samtools mpileup -C 11 -B -q0 -Q0 -A -f x.ref x.sam 2>/dev/null | egrep '\^'
x       1       A       1       ^IT     H
x       2       A       2       C^IT    IH
jkb@seq3b[samtools.../samtools] ./samtools_BAQ mpileup -C 11 -B -q0 -Q0 -A -f x.ref x.sam 2>/dev/null | egrep '\^'
```
-C 11 throws away read 3-5 but leaves 1 and 2.

The samtools_BAQ is built from the mpileup_BAQ branch, see #407.  It appears to fix this bug by luck, as the BAQ calculation being fixed happens to be adjacent to the bam_cap_mapQ call and so that is fixed for free.  It would appear (although unchecked), that the bam_prob_realn_core call in the same function means -E is fixed as well.

Ie fixed by #407.
